### ISSUE TITLE
[Bug] TypeError: Cannot read properties of undefined (reading 'layout')

### ISSUE THREAD

**Issue by farstrider** [benphelps/homepage#564]
### Description

After the November 27th update, all the services on my homepage disappeared and the default layout is shown.  I checked all the configuration files, and they are still the same as they were before the update.

### Steps to reproduce

1. Load homepage in browser
2. See that the default layout/services is shown rather than my configured services

### homepage version

v0.5.1 (f16bd91, Nov 27, 2022)

### Installation method

Docker

### Configuration

```yaml
My services.yaml file looks like this:

- Admin:
    - pfSense:
        icon: pfsense.png
        href: http://192.168.10.1/

    - TrueNAS:
        icon: truenas-scale.png
        href: http://truenas.home.farstrider.eu

    - Aruba Instant On:
        icon: /icons/instanton.png
        href: https://portal.arubainstanton.com/#/login

    - Portainer:
        icon: portainer.png
        href: https://portainer.home.farstrider.eu
        server: docker
        container: portainer
        widget:
          type: portainer
          url: https://portainer.home.farstrider.eu
          env: 2
          key: <key>

    - Nginx Proxy Manager:
        icon: nginxproxymanager.png
        href: http://proxymgr.home.farstrider.eu
        server: docker
        container: NginxProxyManager
        widget:
          type: npm
          url: http://proxymgr.home.farstrider.eu
          username: <username>
          password: <password>

- Network:
    - ns01:
        icon: pihole.png
        href: http://ns01.home.farstrider.eu/admin
        widget:
          type: pihole
          url: http://ns01.home.farstrider.eu/

    - ns02:
        icon: pihole.png
        href: http://ns02.home.farstrider.eu/admin
        widget:
          type: pihole
          url: http://ns02.home.farstrider.eu/

    - phpIPAM:
        icon: phpipam.png
        href: http://ipam.home.farstrider.eu
        server: docker
        container: phpipam-phpipam-web-1

    - netbox:
        icon: netbox.png
        href: http://netbox.home.farstrider.eu
        server: docker
        container: netbox

- Media:
    - Sonarr:
        icon: sonarr.png
        href: http://sonarr.home.farstrider.eu
        server: docker
        container: sonarr
        widget:
          type: sonarr
          url: http://sonarr.home.farstrider.eu
          key: <key>

    - Radarr:
        icon: radarr.png
        href: http://radarr.home.farstrider.eu
        server: docker
        container: radarr
        widget:
          type: radarr
          url: http://radarr.home.farstrider.eu
          key: <key>

    - Prowlarr:
        icon: prowlarr.png
        href: http://prowlarr.home.farstrider.eu
        server: docker
        container: prowlarr
        widget:
          type: prowlarr
          url: http://prowlarr.home.farstrider.eu
          key: <key>

    - Bazarr:
        icon: bazarr.png
        href: http://bazarr.home.farstrider.eu
        server: docker
        container: bazarr
        widget:
          type: bazarr
          url: http://bazarr.home.farstrider.eu
          key: <key>

    - Jellyfin:
        icon: jellyfin.png
        href: http://jellyfin.home.farstrider.eu
        server: docker
        container: jellyfin
        widget:
          type: jellyfin
          url: http://jellyfin.home.farstrider.eu
          key: <key>

- Downloaders:
    - qbittorrent:
        icon: qbittorrent.png
        href: http://qbit.home.farstrider.eu
        server: docker
        container: qBittorrent
#        widget:
#          type: qbittorrent
#          url: http://192.168.50.6:8080
#          username: <username>
#          password: <password>

    - SABnzbd:
        icon: sabnzbd.png
        href: http://sabnzbd.home.farstrider.eu
        server: docker
        container: sabnzbd
        widget:
          type: sabnzbd
          url: http://sabnzbd.home.farstrider.eu
          key: <key>

- Development:
    - GitLab:
        icon: gitlab.png
        href: http://gitlab.home.farstrider.eu
        server: docker
        container: gitlab

    - Wordpress:
        icon: wordpress.png
        href: http://wordpress.home.farstrider.eu
        server: docker
        container: wordpress

- Data Visualization:
    - Grafana:
        icon: grafana.png
        href: http://grafana.home.farstrider.eu
        server: docker
        container: grafana

    - Chronograf:
        icon: chronograf.png
        href: http://db.home.farstrider.eu:8888

- Public Apps:
    - ownCloud:
        icon: owncloud.png
        href: <public url>
        server: docker
        container: owncloud_server

    - Bitwarden:
        icon: bitwarden.png
        href: <public url>
        server: docker
        container: VaultWarden
```

### Container Logs

TypeError: Cannot read properties of undefined (reading 'layout')
    at servicesResponse (/app/.next/server/chunks/8317.js:98:42)
    at async handler (/app/.next/server/pages/api/services.js:56:12)
    at async Object.apiResolver (/app/node_modules/.pnpm/next@12.3.1_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/api-utils/node.js:366:9)
    at async NextNodeServer.runApi (/app/node_modules/.pnpm/next@12.3.1_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/next-server.js:481:9)
    at async Object.fn (/app/node_modules/.pnpm/next@12.3.1_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/next-server.js:735:37)
    at async Router.execute (/app/node_modules/.pnpm/next@12.3.1_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/router.js:247:36)
    at async NextNodeServer.run (/app/node_modules/.pnpm/next@12.3.1_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/base-server.js:347:29)
    at async NextNodeServer.handleRequest (/app/node_modules/.pnpm/next@12.3.1_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/base-server.js:285:20)
    at async Server.<anonymous> (/app/server.js:19:5)

**Comment by mzrimsek** [comment]
Just wanted to chime in to say I can confirm I am seeing similar issues on my end as well. After updating to the latest container image yesterday evening my layout settings seem to be getting ignored.

**Comment by rohan3au** [comment]
I am also facing this issue, I found just updating the layout (change column qty or something) saving, deploying and it then starts reading the file? You can then change back and it is fixed.

**Comment by farstrider** [comment]
> I am also facing this issue, I found just updating the layout (change column qty or something) saving, deploying and it then starts reading the file? You can then change back and it is fixed.

Confirmed.  As long as their is a "layout" section defined in settings.yaml, there is no issue.  However, if settings.yaml is empty, the error comes back.

**Comment by shamoon** [comment]
Someone else reported this but I cannot reproduce it. If you don’t have a layout block things should be the same. If you do, the ordering etc will come from the layout block. 

Have folks here tried recreating their containers?

The relevant line is here: https://github.com/benphelps/homepage/blob/41639752e873c4b96cf1c7e6ecd51d9dd1d9321f/src/utils/config/api-response.js#L81

**Comment by shamoon** [comment]
Only other thing would be if no settings were found at all, perhaps a runtime thing, I can imagine a fix if that’s the case

**Comment by shamoon** [comment]
Do folks here have a _completely_ empty settings.yaml?

**Comment by farstrider** [comment]
> Do folks here have a _completely_ empty settings.yaml?

Yes, my settings.yaml file was completely empty when I got the above error.  I also tried recreating the container, but that had no affect.

**Comment by shamoon** [comment]
If your settings.yaml file is completely empty the app actually throws a bunch of errors, I suspect folks just havent noticed this before. I will release a fix in v5.0.2 for this now

**Comment by mzrimsek** [comment]
> I am also facing this issue, I found just updating the layout (change column qty or something) saving, deploying and it then starts reading the file? You can then change back and it is fixed.

This worked for me. Changed the number of columns for one layout section, saved, reloaded the page and it loaded the layout correctly. Turned back the setting and everything is as it should be now.
